roles/chatbot/templates/chatbot.service.yaml.j2

```diff
@@ -7,17 +7,36 @@ metadata:
   labels:
     {{ lookup("template", "../common/templates/labels/common.yaml.j2")  | indent(width=4) | trim }}
     app.kubernetes.io/component: '{{ deployment_type }}-chatbot-api'
+{% if is_openshift %}
+  annotations:
+    service.beta.openshift.io/serving-cert-secret-name: '{{ ansible_operator_meta.name }}-chatbot-api-tls'
 {% if service_annotations %}
+    {{ service_annotations | indent(width=4) }}
+{% endif %}
+{% elif service_annotations %}
   annotations:
     {{ service_annotations | indent(width=4) }}
 {% endif %}
 spec:
   ports:
+{% if is_openshift %}
+  - port: 8443
+    protocol: TCP
+    targetPort: 8321
+    name: https-api
+{% if enable_http_backward_compatibility | default(false) %}
```

Comment:
This should only be enabled when is_openshift is true, OR you can unconditionally expose the port 8080, and remove the block after..